test/e2e/src/test-non-invoker-signing-by-contracts.js

```diff
@@ -0,0 +1,57 @@
+const { expect } = require("chai");
+const { clientFor } = require("./util");
+
+describe("cross-contract auth", function () {
+  before(async function () {
+    const { contractId: signingContractId, keypair } =
+      await clientFor("doesSigning");
+    const { client: needsSignature } = await clientFor("needsSignature", {
+      keypair,
+    });
+
+    const tx = await needsSignature.hello({
+      to: keypair.publicKey(),
+      user_signer: keypair.publicKey(),
+      contract_signer: signingContractId,
+    });
+    this.context = { tx, signingContractId };
+  });
+
+  describe("needsNonInvokerSigningBy", function () {
+    it("does not assume stellar accounts", async function () {
+      expect(this.context.tx.needsNonInvokerSigningBy()[0]).to.equal(
+        this.context.signingContractId,
+      );
+    });
+  });
+
+  describe("sign", function () {
+    it("doesn't throw error when nonInvokerSigningBy returns a contract", async function () {
+      expect(
+        async () => await this.context.tx.sign({ force: true }),
+      ).to.not.throw();
+    });
+  });
+
+  describe("signAuthEntries with custom authorizeEntry", function () {
+    it("allows signing twice", function () {
+      function authorizeEntry(entry) {
+        return entry;
+      }
+      expect(
+        async () =>
+          await this.context.tx.signAuthEntries({
+            address: this.context.signingContractId,
+            authorizeEntry,
+          }),
+      ).to.not.throw();
+      expect(
+        async () =>
```

Comment:
This is different than the way we do it in other parts of the code since we use the `function (done)` pattern instead of async/await. I've seen in the past that if you screw the pattern up the test looks like it passes but because the async stuff wasn't done correctly it's basically just passing because it didn't run anything. Could you make sure that that's not happening here by doing a client-side `console.log` and sanity checking that you see the output in the success case? Just trying to cover our bases :pray:

Reply:
Switched to `done` approach